### ISSUE TITLE
[PCL] Use C++11 random generators, not `rand`

### ISSUE THREAD

**Issue by kunaltyagi** [PointCloudLibrary/pcl#3781]
**Context**

`rand` has several issues, which C++11 random generators don't. New PR often see the `rand` usage and try to emulate the wrong behavior.

**Expected behavior**

Remove all `rand` so when people search for how PCL did its random num generation, they get good quality results

**Additional context**

[Usage of old-fashioned rand discovered by codacity (or maybe try grep)](https://app.codacy.com/manual/kunaltyagi/pcl/issues?bid=16056128&filters=W3siaWQiOiJMYW5ndWFnZSIsInZhbHVlcyI6W119LHsiaWQiOiJDYXRlZ29yeSIsInZhbHVlcyI6W119LHsiaWQiOiJMZXZlbCIsInZhbHVlcyI6W119LHsiaWQiOiJQYXR0ZXJuIiwidmFsdWVzIjpbIjEzNzcyIl19LHsiaWQiOiJBdXRob3IiLCJ2YWx1ZXMiOltdfV0=)

**Comment by shrijitsingh99** [comment]
There is already `UniformGenerator` in [random.h](https://github.com/PointCloudLibrary/pcl/blob/master/common/include/pcl/common/random.h) and `CloudGenerator` in [generator.h](https://github.com/PointCloudLibrary/pcl/blob/master/common/include/pcl/common/generate.h). 
We should use that only right?

**Comment by kunaltyagi** [comment]
Yes, and maybe add a documentation there to use `random_engine` to generate the seed and not time or any other weird usage (except a constant seed)

**Comment by redfish-lee** [comment]
`rand` has used a lot, especially for testing. 
I'm working on replacing to `UniformGenerator`. But is there a way to eliminate `rand` usage in the future?

**Comment by kunaltyagi** [comment]
> is there a way to eliminate rand usage in the future?

Usage in C++ (in general) or in PCL?

**Comment by redfish-lee** [comment]
In PCL. 
I've dumped files involved with `rand()`, hundreds.
Most of them are found in test files. Perhaps using `rand()` in testing is more convenient and straight-forward for most people. This makes me think if there's a way to avoid using `rand()` in the future.

**Comment by kunaltyagi** [comment]
> in testing is more convenient

We're talking about the PCL source code, not the application code that users of PCL write. Please feel free to use `rand` of the better alternatives provided in PCL: `UniformGenerator` and `CloudGenerator`.
